packages/extension/src/browser/vscode/api/main.thread.treeview.ts

```
@@ -586,6 +588,43 @@ export class TreeViewDataProvider extends Tree {
+          findChildrenToChange(child, nodes);
+        });
+      }
+    }
+
+    const nodesToChange = [node];
+    if (!manageCheckboxStateManually) {
+      findParentsToChange(node, nodesToChange);
+      findChildrenToChange(node, nodesToChange);
+    }
+    nodesToChange.forEach((n) => (n.checkboxInfo!.checked = checked));
+    this.proxy?.$checkStateChanged(this.treeViewId, [{ treeItemId: node.treeItemId, checked }]);
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)